modules/nback.llvm/llvm.bal

```diff
@@ -257,8 +272,27 @@ public class Builder {
     // Corresponds to LLVMBuildCall
     // Returns () if there is no result i.e. function return type is void
     public function call(Function fn, Value[] args) returns Value? {
-        // XXX todo
-        return ();
+        if fn.paramCount() != args.length() {
+            panic error(string `Number of arguments is invalid for function ${fn.ref()}`);
+        }
+        BasicBlock bb = self.bb();
+        string[] argStringContent = [];
+        foreach Value arg in args {
+            argStringContent.push(" ".'join(typeToString(arg.ty), arg.operand));
```

Comment:
Please don't use join here. Instead do it as I did in `header`.

Reply:
Fixed